### PR TITLE
🧑‍💻 More lints to improve and ensure better code quality

### LINKED ISSUE #12: 🧑‍💻 More lints to improve and ensure better code quality
### What would you like to share?

Currently, I find that in the project has a very few lints to check for code quality, which can be important to ensure better code structure and format. Which helps in providing a better developer experience when working on the project.

### Do you want to work on this issue?

Yes

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by neon-mmd**
### Please check if the PR fulfills these requirements

*   [x] The commit message follows our guidelines
*   [x] Tests for the changes have been added (for bug fixes/features)
*   [x] Docs have been added / updated (for bug fixes / features)

### What issue does this fix?
Closes #12     

### What kind of change does this PR introduce?
This PR provides more lints for checking and ensuring code quality, which can help improve developer experience.

### CODE REVIEW

**Comment by Brian3647** [comment]
I don't think a version bump is needed for this

**Comment by Brian3647** [comment]
What do cognitive_complexity and rust_2018_idioms check for?

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "snowboard"
-version = "1.0.3"
+version = "1.0.4"
```
Comment (Brian3647) [inline Cargo.toml:3]: Roll back to 1.0.3 and I'll merge

**Review by Brian3647 — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (neon-mmd) [reply Cargo.toml:3]: Thanks :heart: for taking the time to review this PR. 

Ok sure :slightly_smiling_face: I will revert the change I thought you follow the [semver](semver.org) releases but apparently I think it is not the case. :slightly_smiling_face:

**Comment by neon-mmd** [comment]
> What do cognitive_complexity and rust_2018_idioms check for?

Ok sure, the `cognitive_complexity` check actually checks for things like how complex the code looks like for example if the function is like 1000 lines long or you have something which has a lot of nesting then this check will suggest to split those code into smaller functions which can help maintaining and reading through and following the code alot easier for anyone working on the project.

For the `rust_2018_idioms` check it checks how idiomatic your code is like if it is not then it will suggest some suggestions on how to make it. So that your code looks cleaner and looks more rusty. :slightly_smiling_face:

**Comment by Brian3647** [comment]
All good! Ty for contributing, appreciate it 🙏

**Review by Brian3647 — APPROVED** [review]

**Pull request merged**